src/script/widgets/WMSStylesDialog.js

```diff
@@ -415,6 +441,83 @@ gxp.WMSStylesDialog = Ext.extend(Ext.Container, {
         this.showDlg(styleProperties);
     },
     
+
+    /** api: method[classifyStyleRules]
+     *  :arg prevStyle: ``Ext.data.Record``
+     *
+     *  Edit the currently selected style with classification options enabled.
+     */
+    classifyStyleRules:function () {
+        var userStyle = this.selectedStyle.get("userStyle");
+
+        var rule = userStyle.rules[0];
+        var origRules = [];
+        for (var i=0; i < userStyle.rules.length; i++){
+            origRules[i] = userStyle.rules[i].clone();
+        }
+
+        var ruleDlg = new this.dialogCls({
+            title: String.format(this.ruleWindowTitle,
+                rule.title || rule.name || this.newRuleText),
+            shortTitle: rule.title || rule.name || this.newRuleText,
+            layout: "fit",
+            width: 320,
+            height: 450,
+            modal: true,
+            items: [{
+                xtype: "gxp_rulepanel",
+                ref: "rulePanel",
+                symbolType: this.symbolType,
+                rule: rule,
+                fonts:this.fonts,
+                classifyEnabled: true,
+                attributes: new GeoExt.data.AttributeStore({
```

Comment:
since this is now duplicated code, can you make a function that creates the attribute store instead an reuse that in the two places?